src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -154,27 +159,38 @@ private String buildMessageTitle(EventNotificationContext ctx) {
 
 	String buildCustomMessage(EventNotificationContext ctx, SlackEventNotificationConfig config, String template) throws PermanentEventNotificationException {
 		List<MessageSummary> backlog = getAlarmBacklog(ctx);
-		Map<String, Object> model = getCustomMessageModel(ctx, config, backlog);
-		try {
-			LOG.debug("template = {} model = {}" ,template, model);
-			return templateEngine.transform(template, model);
-		} catch (Exception e) {
-			LOG.error("Exception during templating [{}]", e.toString());
-			throw new PermanentEventNotificationException(e.toString(),e.getCause());
+		backlog = getMessageBacklog(config, backlog);
+
+		Map<String, Object> model = getCustomMessageModel(ctx, config.type(), backlog);
+			try {
+				LOG.debug("template = {} model = {}", template, model);
+				return templateEngine.transform(template, model);
+			} catch (Exception e) {
+				LOG.error("Exception during templating [{}]", e.toString());
+				throw new PermanentEventNotificationException(e.toString(), e.getCause());
+		}
+
+
+	}
+
+	List<MessageSummary> getMessageBacklog(SlackEventNotificationConfig config, List<MessageSummary> backlog) {
+		if(config.backlogSize() > 0 && backlog != null) {
+			backlog = backlog.stream().limit(config.backlogSize()).collect(Collectors.toList());
 		}
+		return backlog;
 	}
 
 
 	List<MessageSummary> getAlarmBacklog(EventNotificationContext ctx) {
 		return notificationCallbackService.getBacklogForEvent(ctx);
 	}
 
-	Map<String, Object> getCustomMessageModel(EventNotificationContext ctx, SlackEventNotificationConfig config, List<MessageSummary> backlog) {
+	Map<String, Object> getCustomMessageModel(EventNotificationContext ctx, String type, List<MessageSummary> backlog) {
```

Comment:
What's the reason for changing the parameter from `SlackEventNotificationConfig` to `String`? This doesn't seem to be related to the backlog change.